### PR TITLE
Testing and linting

### LINKED ISSUE #8: TODO: Add tests
Would be good to at least test that every component has the required `__x__` attributes

### PR DESCRIPTION
**Pull request by blackary**
Fixes #8

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
 streamlit
 Unidecode
 streamlit-keyup>=0.1.6
+st-annotated-text
```
Comment (arnaudmiribel) [inline requirements.txt:4]: Is this file needed? Given we have Pipfile?
If yes it's missing the streamlit-embed-code right?

File: pyproject.toml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools>=61.0", "streamlit==1.11.1", "htbuilder"]
+requires = ["setuptools>=61.0", "streamlit==1.11.1"]
```
Comment (arnaudmiribel) [inline pyproject.toml:2]: 🙏

File: pyproject.toml [hunk 3]
```diff
@@ -21,7 +21,44 @@ keywords = [
     "ui",
     "data",
 ]
+dependencies = [
+  "htbuilder",
+  "st-annotated-text",
+  "streamlit-embedcode",
+  "streamlit-keyup"
+]
```
Comment (arnaudmiribel) [inline pyproject.toml:29]: I think there's a way to just read from Pipfile. But let's go ahead with that.

Q: Isn't that missing `streamlit`? I guess it's included by transitivity, though...

File: Pipfile [hunk 4]
```diff
@@ -11,6 +11,7 @@ st-annotated-text = "*"
 streamlit-embedcode = "*"
 
 [dev-packages]
+pytest = "*"
```
Comment (arnaudmiribel) [inline Pipfile:14]: Shouldn't there be more here? Like pre-commit, black, etc?

File: tests/test_extras.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    if hasattr(mod, "__package_name__"):
+        assert type(mod.__package_name__) == str
+    if hasattr(mod, "__streamlit_cloud_url__"):
+        assert type(mod.__streamlit_cloud_url__) == str
+    if hasattr(mod, "__twitter_username__"):
+        assert type(mod.__twitter_username__) == str
+    if hasattr(mod, "__buymeacoffee_username__"):
+        assert type(mod.__buymeacoffee_username__) == str
+    if hasattr(mod, "__experimental_playground__"):
+        assert type(mod.__experimental_playground__) == bool
```
Comment (arnaudmiribel) [inline tests/test_extras.py:43]: ❤️‍🔥

**Review by arnaudmiribel — COMMENTED** [review]
Bunch of comments on intersection between Pipfile and pyproject.toml... but otherwise LGTM.

File: requirements.txt [hunk 6]
```diff
@@ -1,3 +1,4 @@
 streamlit
 Unidecode
```
Comment (arnaudmiribel) [inline requirements.txt:2]: Also not sure if we need that boy

File: requirements.txt (hunk 1 above)
Comment (sfc-gh-zblackwood) [reply requirements.txt:4]: Yeah, we probably don't need it.

File: Pipfile (hunk 4 above)
Comment (blackary) [reply Pipfile:14]: Yeah, there should.
